{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data wrangling for *Mtb*-Tn-Mat project"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* reads in SI tables / files and compiles them into Pandas dataframe \n",
    "* version 2.0: goal is to have 3 matrices:\n",
    "    1. log2FC matrix\n",
    "    2. q-value matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports modules:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import os\n",
    "import numpy as np\n",
    "from statsmodels.stats.multitest import multipletests\n",
    "from Tn_data_wrangling import *\n",
    "\n",
    "%load_ext autoreload\n",
    "%autoreload 2\n",
    "\n",
    "pd.options.mode.chained_assignment = None  # default='warn"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Read in full set of Rv_IDs and gene_names from tuberculist annotation: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "root_dir = '../../'\n",
    "data_path = os.path.join(root_dir, 'data')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rv_ID</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Rv0001</td>\n",
       "      <td>dnaA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Rv_ID gene_name\n",
       "0  Rv0001      dnaA"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file_WG = 'annotations/H37Rv_tuberculist_functional_categories.xlsx'\n",
    "df_WG = pd.read_excel(os.path.join(data_path, file_WG))\n",
    "df_WG = df_WG[['Rv_ID', 'gene_name']]\n",
    "df_WG.head(1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## The first set of cases:\n",
    "\n",
    "These are the SI files and tables (19 columns) that are in the following format:\n",
    "\n",
    "they list the set of genes that are called (conditionally) essential in the Tn screen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file</th>\n",
       "      <th>col_name</th>\n",
       "      <th>q_val_col_name</th>\n",
       "      <th>ratio_col_name</th>\n",
       "      <th>is_ratio_log2FC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2003_Sassetti_Boyd_Rubin/table_1.xls</td>\n",
       "      <td>2003A_Sassetti</td>\n",
       "      <td>p_val</td>\n",
       "      <td>ratio</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2003_Sassetti_Rubin/table_3.xls</td>\n",
       "      <td>2003B_Sassetti</td>\n",
       "      <td>Week_8_P_value</td>\n",
       "      <td>Week_8_Ratio</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2005_Rengarajan_Rubin/table_2.xls</td>\n",
       "      <td>2005_Rengarajan</td>\n",
       "      <td>Nan</td>\n",
       "      <td>Nan</td>\n",
       "      <td>Nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2006_Joshi_Sassetti/table_1.xlsx</td>\n",
       "      <td>2006_Joshi_GI_1</td>\n",
       "      <td>Nan</td>\n",
       "      <td>Nan</td>\n",
       "      <td>Nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2006_Joshi_Sassetti/table_2.xlsx</td>\n",
       "      <td>2006_Joshi_GI_2</td>\n",
       "      <td>Nan</td>\n",
       "      <td>Nan</td>\n",
       "      <td>Nan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2011_Griffin_Sassetti/table_4.xlsx</td>\n",
       "      <td>2011_Griffin_2</td>\n",
       "      <td>p value</td>\n",
       "      <td>ratio</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2013_Zhang_Rubin/table_2.xlsx</td>\n",
       "      <td>2013_Zhang_1A</td>\n",
       "      <td>FDR (q-val) Day 10</td>\n",
       "      <td>in vitro:in vivo ratio Day 10</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2013_Zhang_Rubin/table_2.xlsx</td>\n",
       "      <td>2013_Zhang_1B</td>\n",
       "      <td>FDR (q-val) Day 45</td>\n",
       "      <td>in vitro:in vivo ratio Day 45</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2013_Zhang_Rubin/table_3.xlsx</td>\n",
       "      <td>2013_Zhang_2</td>\n",
       "      <td>MWU P-value</td>\n",
       "      <td>wt:MHC ratio</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2013_Zhang_Rubin/table_4A.xlsx</td>\n",
       "      <td>2013_Zhang_3A</td>\n",
       "      <td>MWU P-value</td>\n",
       "      <td>Ty6.5:Ty4.5 ratio</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2013_Zhang_Rubin/table_4B.xlsx</td>\n",
       "      <td>2013_Zhang_3B</td>\n",
       "      <td>MWU P-value</td>\n",
       "      <td>Ty6.5:pcit4.5 ratio</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2013_Zhang_Rubin/table_4C.xlsx</td>\n",
       "      <td>2013_Zhang_3C</td>\n",
       "      <td>MWU P-value</td>\n",
       "      <td>pH7:DETANO ratio</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2013_Zhang_Rubin/table_4D.xlsx</td>\n",
       "      <td>2013_Zhang_3D</td>\n",
       "      <td>MWU P-value</td>\n",
       "      <td>trp:Rv ratio</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2016_Korte_Kalscheuer/table_1.xlsx</td>\n",
       "      <td>2016_Korte</td>\n",
       "      <td>p-adj</td>\n",
       "      <td>Nan</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2018_Rittershaus_Sassetti/table_2A.xlsx</td>\n",
       "      <td>2018_Rittershaus_1A</td>\n",
       "      <td>Q Value</td>\n",
       "      <td>Log2 Fold Change (H6/Input)</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2018_Rittershaus_Sassetti/table_2B.xlsx</td>\n",
       "      <td>2018_Rittershaus_1B</td>\n",
       "      <td>Q Value</td>\n",
       "      <td>Log2 Fold Change (H3/Input)</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       file             col_name  \\\n",
       "0      2003_Sassetti_Boyd_Rubin/table_1.xls       2003A_Sassetti   \n",
       "1           2003_Sassetti_Rubin/table_3.xls       2003B_Sassetti   \n",
       "2         2005_Rengarajan_Rubin/table_2.xls      2005_Rengarajan   \n",
       "3          2006_Joshi_Sassetti/table_1.xlsx      2006_Joshi_GI_1   \n",
       "4          2006_Joshi_Sassetti/table_2.xlsx      2006_Joshi_GI_2   \n",
       "5        2011_Griffin_Sassetti/table_4.xlsx       2011_Griffin_2   \n",
       "6             2013_Zhang_Rubin/table_2.xlsx        2013_Zhang_1A   \n",
       "7             2013_Zhang_Rubin/table_2.xlsx        2013_Zhang_1B   \n",
       "8             2013_Zhang_Rubin/table_3.xlsx         2013_Zhang_2   \n",
       "9            2013_Zhang_Rubin/table_4A.xlsx        2013_Zhang_3A   \n",
       "10           2013_Zhang_Rubin/table_4B.xlsx        2013_Zhang_3B   \n",
       "11           2013_Zhang_Rubin/table_4C.xlsx        2013_Zhang_3C   \n",
       "12           2013_Zhang_Rubin/table_4D.xlsx        2013_Zhang_3D   \n",
       "13       2016_Korte_Kalscheuer/table_1.xlsx           2016_Korte   \n",
       "14  2018_Rittershaus_Sassetti/table_2A.xlsx  2018_Rittershaus_1A   \n",
       "15  2018_Rittershaus_Sassetti/table_2B.xlsx  2018_Rittershaus_1B   \n",
       "\n",
       "        q_val_col_name                 ratio_col_name is_ratio_log2FC  \n",
       "0                p_val                          ratio           FALSE  \n",
       "1       Week_8_P_value                   Week_8_Ratio           FALSE  \n",
       "2                  Nan                            Nan             Nan  \n",
       "3                  Nan                            Nan             Nan  \n",
       "4                  Nan                            Nan             Nan  \n",
       "5              p value                          ratio           FALSE  \n",
       "6   FDR (q-val) Day 10  in vitro:in vivo ratio Day 10           FALSE  \n",
       "7   FDR (q-val) Day 45  in vitro:in vivo ratio Day 45           FALSE  \n",
       "8          MWU P-value                   wt:MHC ratio           FALSE  \n",
       "9          MWU P-value              Ty6.5:Ty4.5 ratio           FALSE  \n",
       "10         MWU P-value            Ty6.5:pcit4.5 ratio           FALSE  \n",
       "11         MWU P-value               pH7:DETANO ratio           FALSE  \n",
       "12         MWU P-value                   trp:Rv ratio           FALSE  \n",
       "13               p-adj                            Nan             NaN  \n",
       "14             Q Value    Log2 Fold Change (H6/Input)            TRUE  \n",
       "15             Q Value    Log2 Fold Change (H3/Input)            TRUE  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in path/file_name to column name mapping for 'simple' cases:\n",
    "col_name_file = os.path.join( data_path, 'SI_datasets/files_and_columns.csv') \n",
    "df_col_info_set1 = pd.read_csv(col_name_file)\n",
    "df_col_info_set1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Building q-value and log2FC matrix for first set of files: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'../../data/SI_datasets'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tn_data_path = os.path.join(root_dir, 'data/SI_datasets')\n",
    "tn_data_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2003_Sassetti_Boyd_Rubin/table_1.xls 2003A_Sassetti\n",
      "2003_Sassetti_Rubin/table_3.xls 2003B_Sassetti\n",
      "2005_Rengarajan_Rubin/table_2.xls 2005_Rengarajan\n",
      "2006_Joshi_Sassetti/table_1.xlsx 2006_Joshi_GI_1\n",
      "2006_Joshi_Sassetti/table_2.xlsx 2006_Joshi_GI_2\n",
      "2011_Griffin_Sassetti/table_4.xlsx 2011_Griffin_2\n",
      "2013_Zhang_Rubin/table_2.xlsx 2013_Zhang_1A\n",
      "2013_Zhang_Rubin/table_2.xlsx 2013_Zhang_1B\n",
      "2013_Zhang_Rubin/table_3.xlsx 2013_Zhang_2\n",
      "2013_Zhang_Rubin/table_4A.xlsx 2013_Zhang_3A\n",
      "2013_Zhang_Rubin/table_4B.xlsx 2013_Zhang_3B\n",
      "2013_Zhang_Rubin/table_4C.xlsx 2013_Zhang_3C\n",
      "2013_Zhang_Rubin/table_4D.xlsx 2013_Zhang_3D\n",
      "2016_Korte_Kalscheuer/table_1.xlsx 2016_Korte\n",
      "2018_Rittershaus_Sassetti/table_2A.xlsx 2018_Rittershaus_1A\n",
      "2018_Rittershaus_Sassetti/table_2B.xlsx 2018_Rittershaus_1B\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/ajinich/Documents/git_repos/mtb_tn_db/code/data_wrangling/Tn_data_wrangling.py:43: RuntimeWarning: divide by zero encountered in log2\n",
      "  df_tn[target_log2FC_col] = np.log2(df_tn[log2FC_col])\n"
     ]
    }
   ],
   "source": [
    "counter = 0\n",
    "\n",
    "for index, row in df_col_info_set1.iterrows():\n",
    "    \n",
    "    file_in = row['file']\n",
    "    col_name = row['col_name']\n",
    "    qval_col = row['q_val_col_name']\n",
    "    log2FC_col = row['ratio_col_name']\n",
    "    is_ratio_log2FC = row['is_ratio_log2FC']\n",
    "    target_qval_col = col_name+'_q_val'\n",
    "    target_log2FC_col = col_name+'_log2FC'\n",
    "    \n",
    "    print(file_in, col_name)\n",
    "    \n",
    "    # get column names (both from SI file and target column names in Tn-Matrix )\n",
    "#     qval_col, log2FC_col, is_ratio_log2FC, target_qval_col, target_log2FC_col = get_col_names(file_in, df_col_info_set1, col_name)\n",
    "    \n",
    "    # get data from SI file and populate dataframe. \n",
    "    df_tn_qval_log2FC = build_q_val_ratio_df( os.path.join(tn_data_path, file_in), col_name, qval_col, log2FC_col, is_ratio_log2FC )\n",
    "    \n",
    "    # merge with whole genome (Rv-IDs and gene-names for all of H37Rv)\n",
    "    df_tn_qval_log2FC_WG = merge_with_whole_genome_qval_log2FC( df_WG, target_qval_col, target_log2FC_col, df_tn_qval_log2FC )\n",
    "\n",
    "    # merge with rest of data-files / columns. \n",
    "    if counter == 0:\n",
    "        df_tn_qval_log2FC_ALL = df_tn_qval_log2FC_WG.copy()\n",
    "    else:\n",
    "        df_tn_qval_log2FC_ALL = df_tn_qval_log2FC_ALL.merge(df_tn_qval_log2FC_WG, how = 'inner', on = ['Rv_ID', 'gene_name'])\n",
    "    \n",
    "    counter +=1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Zhang log2FC values are wrong. They report in vitro / in vivo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols_to_invert = ['2013_Zhang_1A_log2FC', '2013_Zhang_1B_log2FC', '2013_Zhang_2_log2FC' ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "for col in cols_to_invert:\n",
    "    df_tn_qval_log2FC_ALL[col] = -df_tn_qval_log2FC_ALL[col]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rv_ID</th>\n",
       "      <th>2013_Zhang_1B_q_val</th>\n",
       "      <th>2013_Zhang_1B_log2FC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Rv0001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Rv0002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rv0003</td>\n",
       "      <td>0.000087</td>\n",
       "      <td>-3.223423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Rv0004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Rv0005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Rv0006</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Rv0007</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>-8.927778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Rv0008c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Rv0009</td>\n",
       "      <td>0.000725</td>\n",
       "      <td>-4.461704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Rv0010c</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Rv_ID  2013_Zhang_1B_q_val  2013_Zhang_1B_log2FC\n",
       "0   Rv0001                  NaN                   NaN\n",
       "1   Rv0002                  NaN                   NaN\n",
       "2   Rv0003             0.000087             -3.223423\n",
       "3   Rv0004                  NaN                   NaN\n",
       "4   Rv0005                  NaN                   NaN\n",
       "5   Rv0006                  NaN                   NaN\n",
       "6   Rv0007             0.000007             -8.927778\n",
       "7  Rv0008c                  NaN                   NaN\n",
       "8   Rv0009             0.000725             -4.461704\n",
       "9  Rv0010c                  NaN                   NaN"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_tn_qval_log2FC_ALL[['Rv_ID', '2013_Zhang_1B_q_val', '2013_Zhang_1B_log2FC']].head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Second set of cases: \n",
    "\n",
    "These are SI datafiles and tables that have, for all genes in H37Rv, a column for q-vals and ratios."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load spreasheet with file name-to-column mappings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>file</th>\n",
       "      <th>col_name</th>\n",
       "      <th>q_val_col_name</th>\n",
       "      <th>ratio_col_name</th>\n",
       "      <th>is_ratio_log2FC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2012_Zhang_Rubin/table_4_sheet_600bp.xlsx</td>\n",
       "      <td>2012_Zhang</td>\n",
       "      <td>p-val avg</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2015_Kieser_Rubin/table_1A_edited.xlsx</td>\n",
       "      <td>2015_Kieser_GI_1</td>\n",
       "      <td>P-value</td>\n",
       "      <td>Fold Change</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2015_Kieser_Rubin/table_1B_edited.xlsx</td>\n",
       "      <td>2015_Kieser_GI_2</td>\n",
       "      <td>P-value</td>\n",
       "      <td>Fold Change</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2015_Kieser_Rubin/table_1C_edited.xlsx</td>\n",
       "      <td>2015_Kieser_GI_3</td>\n",
       "      <td>P-value</td>\n",
       "      <td>Fold Change</td>\n",
       "      <td>FALSE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2015_Mendum_Stewart/table_1A.xlsx</td>\n",
       "      <td>2015_Mendum</td>\n",
       "      <td>p_val_7days</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2016_Nambi_Sassetti/table_1.xlsx</td>\n",
       "      <td>2016_Nambi</td>\n",
       "      <td>Qval</td>\n",
       "      <td>Log2 ctpC/WT</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2017_Mishra_Sassetti/table_1.xlsx</td>\n",
       "      <td>2017_Mishra_1</td>\n",
       "      <td>Qval (C3HeBvsBL6)</td>\n",
       "      <td>log2(C3HeB/BL6)</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2017_Mishra_Sassetti/table_1.xlsx</td>\n",
       "      <td>2017_Mishra_2</td>\n",
       "      <td>Qval (NOS2vsBL6)</td>\n",
       "      <td>log2(NOS2/BL6)</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2017_Xu_Ehrt/table_3A.xlsx</td>\n",
       "      <td>2017_Xu_1A</td>\n",
       "      <td>qval</td>\n",
       "      <td>FC [log2]</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2017_Xu_Ehrt/table_3B.xlsx</td>\n",
       "      <td>2017_Xu_1B</td>\n",
       "      <td>qval</td>\n",
       "      <td>FC [log2]</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2017_Xu_Ehrt/table_3C.xlsx</td>\n",
       "      <td>2017_Xu_1C</td>\n",
       "      <td>qval</td>\n",
       "      <td>FC [log2]</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2017_Xu_Ehrt/table_3D.xlsx</td>\n",
       "      <td>2017_Xu_1D</td>\n",
       "      <td>qval</td>\n",
       "      <td>FC [log2]</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2017_Xu_Ehrt/table_3E.xlsx</td>\n",
       "      <td>2017_Xu_1E</td>\n",
       "      <td>qval</td>\n",
       "      <td>FC [log2]</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2018_Carey_Fortune/table_1A.xlsx</td>\n",
       "      <td>2018_Carey_1A</td>\n",
       "      <td>p-adj</td>\n",
       "      <td>log2 FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2018_Carey_Fortune/table_1B.xlsx</td>\n",
       "      <td>2018_Carey_1B</td>\n",
       "      <td>p-adj</td>\n",
       "      <td>log2 FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2018_Carey_Fortune/table_1C.xlsx</td>\n",
       "      <td>2018_Carey_1C</td>\n",
       "      <td>p-adj</td>\n",
       "      <td>log2 FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2018_Carey_Fortune/table_1D.xlsx</td>\n",
       "      <td>2018_Carey_1D</td>\n",
       "      <td>p-adj</td>\n",
       "      <td>log2 FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2018_Carey_Fortune/table_1E.xlsx</td>\n",
       "      <td>2018_Carey_1E</td>\n",
       "      <td>p-adj</td>\n",
       "      <td>log2 FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2018_Carey_Fortune/table_1F.xlsx</td>\n",
       "      <td>2018_Carey_1F</td>\n",
       "      <td>p-adj</td>\n",
       "      <td>log2 FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2018_Carey_Fortune/table_1G.xlsx</td>\n",
       "      <td>2018_Carey_1G</td>\n",
       "      <td>p-adj</td>\n",
       "      <td>log2 FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>2018_Carey_Fortune/table_1H.xlsx</td>\n",
       "      <td>2018_Carey_1H</td>\n",
       "      <td>p-adj</td>\n",
       "      <td>log2 FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>2017B_DeJesus_Iorger/table_1A.xlsx</td>\n",
       "      <td>2017B_DeJesus_1A</td>\n",
       "      <td>BFDR</td>\n",
       "      <td>Mean delta log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>2017B_DeJesus_Iorger/table_1B.xlsx</td>\n",
       "      <td>2017B_DeJesus_1B</td>\n",
       "      <td>BFDR</td>\n",
       "      <td>Mean delta log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>2017B_DeJesus_Iorger/table_1C.xlsx</td>\n",
       "      <td>2017B_DeJesus_1C</td>\n",
       "      <td>BFDR</td>\n",
       "      <td>Mean delta log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_WT_marP...</td>\n",
       "      <td>marP</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...</td>\n",
       "      <td>Rv0950</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...</td>\n",
       "      <td>Rv3916c</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...</td>\n",
       "      <td>Rv0307c</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_v...</td>\n",
       "      <td>Rv3717</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_g...</td>\n",
       "      <td>ponA1</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...</td>\n",
       "      <td>Rv1565c</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...</td>\n",
       "      <td>Rv3684</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_WT_Rv09...</td>\n",
       "      <td>Rv0954</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_v...</td>\n",
       "      <td>Rv3811</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...</td>\n",
       "      <td>Rv1096</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...</td>\n",
       "      <td>Rv2680</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...</td>\n",
       "      <td>Rv1432</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_v...</td>\n",
       "      <td>Rv3594</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...</td>\n",
       "      <td>Rv3005c</td>\n",
       "      <td>Adj. p-value</td>\n",
       "      <td>log2FC</td>\n",
       "      <td>TRUE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>2013_DeJesus_Iorger/table_1.xls</td>\n",
       "      <td>2013_DeJesus</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>2017A_DeJesus_Iorger/table_1.xlsx</td>\n",
       "      <td>2017A_DeJesus</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "      <td>NONE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 file          col_name  \\\n",
       "0           2012_Zhang_Rubin/table_4_sheet_600bp.xlsx        2012_Zhang   \n",
       "1              2015_Kieser_Rubin/table_1A_edited.xlsx  2015_Kieser_GI_1   \n",
       "2              2015_Kieser_Rubin/table_1B_edited.xlsx  2015_Kieser_GI_2   \n",
       "3              2015_Kieser_Rubin/table_1C_edited.xlsx  2015_Kieser_GI_3   \n",
       "4                   2015_Mendum_Stewart/table_1A.xlsx       2015_Mendum   \n",
       "5                    2016_Nambi_Sassetti/table_1.xlsx        2016_Nambi   \n",
       "6                   2017_Mishra_Sassetti/table_1.xlsx     2017_Mishra_1   \n",
       "7                   2017_Mishra_Sassetti/table_1.xlsx     2017_Mishra_2   \n",
       "8                          2017_Xu_Ehrt/table_3A.xlsx        2017_Xu_1A   \n",
       "9                          2017_Xu_Ehrt/table_3B.xlsx        2017_Xu_1B   \n",
       "10                         2017_Xu_Ehrt/table_3C.xlsx        2017_Xu_1C   \n",
       "11                         2017_Xu_Ehrt/table_3D.xlsx        2017_Xu_1D   \n",
       "12                         2017_Xu_Ehrt/table_3E.xlsx        2017_Xu_1E   \n",
       "13                   2018_Carey_Fortune/table_1A.xlsx     2018_Carey_1A   \n",
       "14                   2018_Carey_Fortune/table_1B.xlsx     2018_Carey_1B   \n",
       "15                   2018_Carey_Fortune/table_1C.xlsx     2018_Carey_1C   \n",
       "16                   2018_Carey_Fortune/table_1D.xlsx     2018_Carey_1D   \n",
       "17                   2018_Carey_Fortune/table_1E.xlsx     2018_Carey_1E   \n",
       "18                   2018_Carey_Fortune/table_1F.xlsx     2018_Carey_1F   \n",
       "19                   2018_Carey_Fortune/table_1G.xlsx     2018_Carey_1G   \n",
       "20                   2018_Carey_Fortune/table_1H.xlsx     2018_Carey_1H   \n",
       "21                 2017B_DeJesus_Iorger/table_1A.xlsx  2017B_DeJesus_1A   \n",
       "22                 2017B_DeJesus_Iorger/table_1B.xlsx  2017B_DeJesus_1B   \n",
       "23                 2017B_DeJesus_Iorger/table_1C.xlsx  2017B_DeJesus_1C   \n",
       "24  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_WT_marP...              marP   \n",
       "25  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...            Rv0950   \n",
       "26  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...           Rv3916c   \n",
       "27  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...           Rv0307c   \n",
       "28  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_v...            Rv3717   \n",
       "29  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_g...             ponA1   \n",
       "30  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...           Rv1565c   \n",
       "31  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...            Rv3684   \n",
       "32  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_WT_Rv09...            Rv0954   \n",
       "33  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_v...            Rv3811   \n",
       "34  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...            Rv1096   \n",
       "35  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...            Rv2680   \n",
       "36  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...            Rv1432   \n",
       "37  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_v...            Rv3594   \n",
       "38  FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_R...           Rv3005c   \n",
       "39                    2013_DeJesus_Iorger/table_1.xls      2013_DeJesus   \n",
       "40                  2017A_DeJesus_Iorger/table_1.xlsx     2017A_DeJesus   \n",
       "\n",
       "       q_val_col_name     ratio_col_name is_ratio_log2FC  \n",
       "0           p-val avg               NONE            NONE  \n",
       "1             P-value        Fold Change           FALSE  \n",
       "2             P-value        Fold Change           FALSE  \n",
       "3             P-value        Fold Change           FALSE  \n",
       "4         p_val_7days               NONE            NONE  \n",
       "5                Qval       Log2 ctpC/WT            TRUE  \n",
       "6   Qval (C3HeBvsBL6)    log2(C3HeB/BL6)            TRUE  \n",
       "7    Qval (NOS2vsBL6)     log2(NOS2/BL6)            TRUE  \n",
       "8                qval          FC [log2]            TRUE  \n",
       "9                qval          FC [log2]            TRUE  \n",
       "10               qval          FC [log2]            TRUE  \n",
       "11               qval          FC [log2]            TRUE  \n",
       "12               qval          FC [log2]            TRUE  \n",
       "13              p-adj            log2 FC            TRUE  \n",
       "14              p-adj            log2 FC            TRUE  \n",
       "15              p-adj            log2 FC            TRUE  \n",
       "16              p-adj            log2 FC            TRUE  \n",
       "17              p-adj            log2 FC            TRUE  \n",
       "18              p-adj            log2 FC            TRUE  \n",
       "19              p-adj            log2 FC            TRUE  \n",
       "20              p-adj            log2 FC            TRUE  \n",
       "21               BFDR  Mean delta log2FC            TRUE  \n",
       "22               BFDR  Mean delta log2FC            TRUE  \n",
       "23               BFDR  Mean delta log2FC            TRUE  \n",
       "24       Adj. p-value             log2FC            TRUE  \n",
       "25       Adj. p-value             log2FC            TRUE  \n",
       "26       Adj. p-value             log2FC            TRUE  \n",
       "27       Adj. p-value             log2FC            TRUE  \n",
       "28       Adj. p-value             log2FC            TRUE  \n",
       "29       Adj. p-value             log2FC            TRUE  \n",
       "30       Adj. p-value             log2FC            TRUE  \n",
       "31       Adj. p-value             log2FC            TRUE  \n",
       "32       Adj. p-value             log2FC            TRUE  \n",
       "33       Adj. p-value             log2FC            TRUE  \n",
       "34       Adj. p-value             log2FC            TRUE  \n",
       "35       Adj. p-value             log2FC            TRUE  \n",
       "36       Adj. p-value             log2FC            TRUE  \n",
       "37       Adj. p-value             log2FC            TRUE  \n",
       "38       Adj. p-value             log2FC            TRUE  \n",
       "39               NONE               NONE            NONE  \n",
       "40               NONE               NONE            NONE  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "col_name_file = os.path.join( data_path, 'SI_datasets/files_and_columns_set2.csv') \n",
    "df_col_info_set2 = pd.read_csv(col_name_file)\n",
    "df_col_info_set2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Don't do the last two; they are Michael's and in a different format: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_col_info = df_col_info_set2[:-2].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2012_Zhang_Rubin/table_4_sheet_600bp.xlsx \t 2012_Zhang\n",
      "2015_Kieser_Rubin/table_1A_edited.xlsx \t 2015_Kieser_GI_1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/adrian/anaconda3/lib/python3.7/site-packages/pandas/core/series.py:853: RuntimeWarning: divide by zero encountered in log2\n",
      "  result = getattr(ufunc, method)(*inputs, **kwargs)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2015_Kieser_Rubin/table_1B_edited.xlsx \t 2015_Kieser_GI_2\n",
      "2015_Kieser_Rubin/table_1C_edited.xlsx \t 2015_Kieser_GI_3\n",
      "2015_Mendum_Stewart/table_1A.xlsx \t 2015_Mendum\n",
      "2016_Nambi_Sassetti/table_1.xlsx \t 2016_Nambi\n",
      "2017_Mishra_Sassetti/table_1.xlsx \t 2017_Mishra_1\n",
      "2017_Mishra_Sassetti/table_1.xlsx \t 2017_Mishra_2\n",
      "2017_Xu_Ehrt/table_3A.xlsx \t 2017_Xu_1A\n",
      "2017_Xu_Ehrt/table_3B.xlsx \t 2017_Xu_1B\n",
      "2017_Xu_Ehrt/table_3C.xlsx \t 2017_Xu_1C\n",
      "2017_Xu_Ehrt/table_3D.xlsx \t 2017_Xu_1D\n",
      "2017_Xu_Ehrt/table_3E.xlsx \t 2017_Xu_1E\n",
      "2018_Carey_Fortune/table_1A.xlsx \t 2018_Carey_1A\n",
      "2018_Carey_Fortune/table_1B.xlsx \t 2018_Carey_1B\n",
      "2018_Carey_Fortune/table_1C.xlsx \t 2018_Carey_1C\n",
      "2018_Carey_Fortune/table_1D.xlsx \t 2018_Carey_1D\n",
      "2018_Carey_Fortune/table_1E.xlsx \t 2018_Carey_1E\n",
      "2018_Carey_Fortune/table_1F.xlsx \t 2018_Carey_1F\n",
      "2018_Carey_Fortune/table_1G.xlsx \t 2018_Carey_1G\n",
      "2018_Carey_Fortune/table_1H.xlsx \t 2018_Carey_1H\n",
      "2017B_DeJesus_Iorger/table_1A.xlsx \t 2017B_DeJesus_1A\n",
      "2017B_DeJesus_Iorger/table_1B.xlsx \t 2017B_DeJesus_1B\n",
      "2017B_DeJesus_Iorger/table_1C.xlsx \t 2017B_DeJesus_1C\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_WT_marP_s10000_pc0.00_1.xlsx \t marP\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv0950_day0_s10000_pc0.00.xlsx \t Rv0950\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv3916c_day0_s10000_pc0.00.xlsx \t Rv3916c\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv0307c_day0_s10000_pc0.00.xlsx \t Rv0307c\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_vs_delta_Rv3717_TTR.xlsx \t Rv3717\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_glycerol_ponA1KO_s10000_pc0.00.xlsx \t ponA1\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv1565c_day32_TTR.xlsx \t Rv1565c\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv3684_day0_s10000_pc0.00.xlsx \t Rv3684\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_WT_Rv0954_s10000_pc0.00.xlsx \t Rv0954\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_vs_delta_Rv3811_TTR.xlsx \t Rv3811\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv1096_day0_s10000_pc0.00.xlsx \t Rv1096\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv2680_day32_s10000_pc0.00.xlsx \t Rv2680\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv1432_day32_TTR.xlsx \t Rv1432\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_vs_delta_Rv3594_TTR.xlsx \t Rv3594\n",
      "FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv3005c_day32_s10000_pc0.00.xlsx \t Rv3005c\n"
     ]
    }
   ],
   "source": [
    "df_tn_qval_log2FC_ALL = get_qval_log2FC_func(df_col_info, df_WG, df_tn_qval_log2FC_ALL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3990, 112)"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_tn_qval_log2FC_ALL.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'../../data'"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "path_out = '../../data/SI_datasets/'\n",
    "out_file = os.path.join(path_out, 'SI_qval_log2FC.xlsx')\n",
    "df_tn_qval_log2FC_ALL.to_excel(out_file, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## checking a couple of gene profiles: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## grabbing only 'full' data columns for Anisha's dash site: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols_set_2 = df_col_info_set2.col_name.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "cols_set_2_qvals_temp = [col+'_q_val' for col in cols_set_2]\n",
    "cols_set_2_qvals = [col for col in cols_set_2_qvals_temp if col in df_tn_qval_log2FC_ALL.columns]\n",
    "\n",
    "\n",
    "cols_set_2_log2FC_temp = [col+'_log2FC' for col in cols_set_2]\n",
    "cols_set_2_log2FC = [col for col in cols_set_2_log2FC_temp if col in df_tn_qval_log2FC_ALL.columns]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_tn_qval_set2 = df_tn_qval_log2FC_ALL[ ['Rv_ID', 'gene_name'] + cols_set_2_qvals]\n",
    "df_tn_log2FC_set2 = df_tn_qval_log2FC_ALL[ ['Rv_ID', 'gene_name'] + cols_set_2_log2FC]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_qval = os.path.join(data_path, 'Tn_library_qvals_DASH.xlsx')\n",
    "df_tn_qval_set2.to_excel(file_qval, index=False)\n",
    "\n",
    "file_log2FC = os.path.join(data_path, 'Tn_library_log2FC_DASH.xlsx')\n",
    "df_tn_log2FC_set2.to_excel(file_log2FC, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### [PENDING] \n",
    "\n",
    "### DeJesus and Iorger (2013)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rv_ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Insertions</th>\n",
       "      <th># TA Sites</th>\n",
       "      <th>Length of Maximum Run</th>\n",
       "      <th>Span of Nucleotides</th>\n",
       "      <th>Sassetti-03</th>\n",
       "      <th>Post. Prob. Of Ess.</th>\n",
       "      <th>Call</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Rv0001</td>\n",
       "      <td>dnaA</td>\n",
       "      <td>1</td>\n",
       "      <td>32</td>\n",
       "      <td>31</td>\n",
       "      <td>1365</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Rv0002</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>31</td>\n",
       "      <td>1167</td>\n",
       "      <td>-1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>E</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Rv_ID  Name  Insertions  # TA Sites  Length of Maximum Run  \\\n",
       "0  Rv0001  dnaA           1          32                     31   \n",
       "1  Rv0002  dnaN           0          31                     31   \n",
       "\n",
       "   Span of Nucleotides  Sassetti-03  Post. Prob. Of Ess. Call  \n",
       "0                 1365            1                  1.0    E  \n",
       "1                 1167           -1                  1.0    E  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file = '2013_DeJesus_Iorger/table_1.xls'\n",
    "col_name = '2013_DeJesus'\n",
    "df_tn = pd.read_excel(os.path.join(dir_name, file))\n",
    "df_tn.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### DeJesus, Iorger (2017A)\n",
    "\n",
    "#####  (you need to talk about this with Michael!!!!)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_list = ['2017A_DeJesus_Iorger/table_1.xlsx']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "____________\n",
    "____________\n",
    "____________\n",
    "# OTHER stuff "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Kieser Rubin (2015 data) - tables A, B, C"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "pre-formatting: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "# file_list = ['2015_Kieser_Rubin/table_1A.xlsx', \n",
    "#              '2015_Kieser_Rubin/table_1B.xlsx',\n",
    "#              '2015_Kieser_Rubin/table_1C.xlsx']\n",
    "\n",
    "# for file in file_list:\n",
    "#     # read the data. \n",
    "#     df_tn = pd.read_excel(os.path.join(dir_name, file))\n",
    "\n",
    "#     # this fixes the Rv_IDs for this particular dataset. \n",
    "#     df_tn = df_tn[~df_tn.Rv_ID.str.contains('IG')]\n",
    "#     rv_id_list = [rv_id.split('_')[-1].strip('\\'') for rv_id in df_tn['Rv_ID'].values]\n",
    "#     df_tn['Rv_ID'] = rv_id_list\n",
    "\n",
    "#     # write edited file:\n",
    "#     outfile = file.split('.xlsx')[0]+'_edited.xlsx'\n",
    "#     df_tn.to_excel(os.path.join(dir_name, outfile), index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### FLUTE datasets pre-processing:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "flute_path = 'Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/original'\n",
    "file_list_flute = [os.path.join(flute_path, file) for file in os.listdir(flute_path)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in file_list_flute:\n",
    "    file_out = file.split('/')[-1].split('.dat')[0]+'.tsv'\n",
    "    file_out = os.path.join('Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/', file_out)\n",
    "\n",
    "    with open(file, 'r') as fin:\n",
    "        lines_temp = fin.read()\n",
    "        lines_new = lines_temp.split('\\n')[6:]\n",
    "        with open(file_out, 'w') as fout:\n",
    "            fout.write('\\n'.join(lines_new))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [],
   "source": [
    "flute_path = 'Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/'\n",
    "tsv_list_flute = [os.path.join(flute_path, file) for file in os.listdir(flute_path) if 'resampling' in file]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_WT_marP_s10000_pc0.00_1.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv0950_day0_s10000_pc0.00.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv3916c_day0_s10000_pc0.00.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv0307c_day0_s10000_pc0.00.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_vs_delta_Rv3717_TTR.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_glycerol_ponA1KO_s10000_pc0.00.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv1565c_day32_TTR.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv3684_day0_s10000_pc0.00.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_WT_Rv0954_s10000_pc0.00.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_vs_delta_Rv3811_TTR.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv1096_day0_s10000_pc0.00.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv2680_day32_s10000_pc0.00.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv1432_day32_TTR.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_vs_delta_Rv3594_TTR.tsv\n",
      "Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv3005c_day32_s10000_pc0.00.tsv\n"
     ]
    }
   ],
   "source": [
    "for file in tsv_list_flute:\n",
    "    print(file)\n",
    "    file_out = file.split('/')[-1].split('.tsv')[0]+'.xlsx'\n",
    "    file_out = os.path.join('Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/', file_out)\n",
    "    df_flute = pd.read_csv(file, sep='\\t')\n",
    "    df_flute.rename(columns={'#Orf':'Rv_ID','Name':'gene_name', 'p-adj':'Adj. p-value', 'log2 FC':'log2FC'}, inplace=True)\n",
    "    df_flute.to_excel(file_out, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [],
   "source": [
    "f_temp = 'Tn_library_studies/FLUTE_KO_TnSeq/FLUTE_log2fc/resampling_H37Rv_Rv1432_day32_TTR.xlsx'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_excel(f_temp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Rv_ID</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Description</th>\n",
       "      <th>N</th>\n",
       "      <th>TAs Hit</th>\n",
       "      <th>Sum Rd 1</th>\n",
       "      <th>Sum Rd 2</th>\n",
       "      <th>Delta Rd</th>\n",
       "      <th>log2FC</th>\n",
       "      <th>p-value</th>\n",
       "      <th>Adj. p-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Rv0001</td>\n",
       "      <td>dnaA</td>\n",
       "      <td>chromosomal replication initiation protein</td>\n",
       "      <td>31</td>\n",
       "      <td>1</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>3.83</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Rv0002</td>\n",
       "      <td>dnaN</td>\n",
       "      <td>DNA polymerase III subunit beta</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rv0003</td>\n",
       "      <td>recF</td>\n",
       "      <td>recombination protein F</td>\n",
       "      <td>35</td>\n",
       "      <td>9</td>\n",
       "      <td>19.1</td>\n",
       "      <td>54.0</td>\n",
       "      <td>34.9</td>\n",
       "      <td>1.50</td>\n",
       "      <td>0.3056</td>\n",
       "      <td>0.69974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Rv0004</td>\n",
       "      <td>-</td>\n",
       "      <td>hypothetical protein Rv0004</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0000</td>\n",
       "      <td>1.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Rv0005</td>\n",
       "      <td>gyrB</td>\n",
       "      <td>DNA gyrase subunit B</td>\n",
       "      <td>42</td>\n",
       "      <td>3</td>\n",
       "      <td>35.2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-34.2</td>\n",
       "      <td>-5.08</td>\n",
       "      <td>0.1396</td>\n",
       "      <td>0.46795</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Rv_ID gene_name                                  Description   N  TAs Hit  \\\n",
       "0  Rv0001      dnaA  chromosomal replication initiation protein   31        1   \n",
       "1  Rv0002      dnaN             DNA polymerase III subunit beta   31        0   \n",
       "2  Rv0003      recF                     recombination protein F   35        9   \n",
       "3  Rv0004         -                 hypothetical protein Rv0004    7        0   \n",
       "4  Rv0005      gyrB                        DNA gyrase subunit B   42        3   \n",
       "\n",
       "   Sum Rd 1  Sum Rd 2  Delta Rd  log2FC  p-value  Adj. p-value  \n",
       "0       0.2       0.0      -0.2    3.83   1.0000       1.00000  \n",
       "1       0.0       0.0       0.0    0.00   1.0000       1.00000  \n",
       "2      19.1      54.0      34.9    1.50   0.3056       0.69974  \n",
       "3       0.0       0.0       0.0    0.00   1.0000       1.00000  \n",
       "4      35.2       1.0     -34.2   -5.08   0.1396       0.46795  "
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
