<HTML>

<H2>Introduction</H2>

The software available here is a python implementation of the Bayesian analysis method referenced above. It utilizes read information obtained from sequencing libraries of transposon mutants, to determine the essentiality of genes. Using a Bayesian framework, essentiality is modeled through the Extreme Value (Gumbel) distribution, which characterizes the maximum run of non-insertions (i.e. number of consecutive TA sites lacking insertion in a row). Genes with significantly larger runs of non-insertion thant statistically expected have a higher likelihood of essentiality. A Metropolis-Hastings sampling procedure is utilized to sample from conditional densities of essentiality for all genes, and posterior estimates of the probability of being essential are estimated for all genes. 

<P>


<H4>Source Code</H4>

Source code is written in Python, and is available online at <A href="http://saclab.tamu.edu/essentiality/">http://saclab.tamu.edu/essentiality/</A>

<H4> <a name="requirements"> Requirements: </a> </H4>

<UL>
	<LI> Python 2.4.3+ <A href="http://www.python.org">www.python.org</A>
	<LI> Scipy 0.6.0+ <A href="http://www.scipy.org/Download">www.scipy.org/Download</A>
	<LI> Numpy 1.2.1+ <A href="http://www.scipy.org/Download">www.scipy.org/Download</A>
</UL>

<BR>
<P>

<H2> <a name="instructions">Instructions</a> </H2>

Before running the python script, please make sure you have installed all the necessary <a href="#requirements">prerequisites</a> listed above, and have downloaded the <a href="http://saclab.tamu.edu/essentiality/#version">latest version</a> of the script. Once the prerequisite software and libraries are installed, to run the script simply type the following command:
<P>
<PRE>
python gumbelMH.py -f test.igv.
</PRE>

Where "-f" is a flag for the input file, and "test.igv" is an example file name. Below the <A href="#format">input file formats</A> are described, followed by description of the <a href="#flags">input flags</a> available, and finally a description of the <a href="#output">output format</a> for the results.

<P>
<BR>

<H4><a name="format">Input File Format</a></H4>

Read data must be contained in a tab-delimited text-file according to the <A href="http://www.broadinstitute.org/software/igv/IGV">IGV format</A>, with two required tracks (i.e. Reads, TAs) representing the read count information of all TA sites in the genome (i.e. with or without insertions), as shown below. The first 5 columns are part of the standard IGV format for representing features in the Integrated Genomic Viewer application. The last two columns represent necessary tracks used by the program to determine the number of reads found at each TA site. TA sites without insertions must be included in the file, and their read counts are represented with zeros. Below is an example of an input file in this format:

<BR>
<BR>
<PRE>
#type=COPY_NUMBER
Chromosome	Start	End	Feature	Reads	TAs
NC_000962.2	59	61	Rv0001	0	1
NC_000962.2	71	73	Rv0001	2	1
NC_000962.2	101	103	Rv0001	36	1
</PRE>

<BR>
<BR>

The table below describes each of the columns necessary for the input file:
<BR>
<BR>
<table>
<TR><TH width="20%" align="left">Column Header</TH> <TH width="80%" align="left">Column Definition<TH></TR>
<TR><TD>Chromosome:</TD> <TD>NCBI Chromosome ID.</TD> </TR>
<TR><TD>Start:</TD> <TD>Start Position of the TA dinucleotides within the genome.</TD></TR>
<TR><TD>End:</TD> <TD>End Position of the TA dinucleotides within the genome.</TD></TR>
<TR><TD>Feature:</TD> <TD>ID of the ORF containing the TA dinucleotide.</TD></TR>
<TR><TD>Reads:</TD> <TD>Read count observed at this particular TA site. Empty sites are represented by zero (0).</TD></TR>
<TR><TD>TAs</TD> <TD>A constant value of "1" for all TA sites. Helpful for to identifying TA sites within the IGV Genomics Viewer.</TD></TR>
</table>

<BR>
<BR>

Example files are provided below to test the execution of the script and help verify that input files are in the approrpiate format:
<UL>
	<LI> <A href="example_reads1.igv">Example File #1</A>
	<LI> <A href="trash_glycerol.igv">Glycerol TraSH Data</A>
</UL>

<BR>

<P>

<H4> <a name="flags">Flags </a> </H4>

The table below outlines the flags accepted by the python script. All but the file flag, "-f", are optional. Default values are described below.
<BR>
<BR>

<table>
<tr> <th width="5%" align="left">Flag</th> <th width="10%" align="left">Value</th> <th width="85%" align="left">Definition</th></tr>
<tr> <td>-f</td> <td>[String]</td> <td>Path to IGV formatted file containing the reads mapped to the genome.</td></tr>


<tr><td>-s</td> <td>[Integer]</td> <td>Desired sample size for the Metropolis Hastings sampling procedure. Default: -s 10000</td></tr>
<tr><td>-m</td> <td>[Integer]</td> <td>Minimum read count to be considered as an "insertion". Default: -m 2</td></tr>
<tr><td>-p</td> <td>[Float]</td> <td>Starting value of the phi parameter (i.e. probability of non-insertion at non-essential genes). Default: -p 0.5</td></tr>
<tr><td>-b</td> <td>[Integer]</td> <td>Number of iterations to be discarded as a Burn-In period. Default: -b 100</td></tr>
<tr><td>-t</td> <td>[Integer]</td> <td>Only every t-th sample is kept. Helps decrease correlation between samples. Default: -t 1</td></tr>
<tr><td>-v</td> <td>N/A</td> <td>Final Sample of essentiality assignments and phi parameter is printed along side results.</td></tr>
</table>

<P>
<BR>

<H4><a name="output">Output Format</a></H4>

Results are printed to screen in tab-separated format:
<BR>
<BR>
<PRE>
#Command used:	python gumbelMH.py -f test.igv -s 10
#MH Acceptance-Rate:	81.06%
#Total Iterations Performed:	301
#Sample Size:	10
#phi estimate:	0.504605
#ORF	k	n	r	s	zbar	sample
Rv0001	1	32	31	1363	1.000000	1,1,1,1,1,1,1,1,1,1
Rv0002	4	8	4	68	0.000000	0,0,0,0,0,0,0,0,0,0
Rv0003	0	8	8	314	1.00000	1,1,1,1,1,1,1,1,1,1
</PRE>
<BR>
Header/Comments are proceded with "#" tags, and are followed by a row for each ORF found in the input file. Below, each of the columns in is defined:

<BR>
<BR>
<table>
<TR><TH width="20%" align="left">Column Header</TH> <TH width="80%" align="left">Column Definition<TH></TR>
<TR><TD>ORF:</TD> <TD>ORF ID found in input file.</TD> </TR>
<TR><TD>k:</TD> <TD>Number of Transposon Insertions Observed within the ORF.</TD></TR>
<TR><TD>n:</TD> <TD>Total Number of TA dinucleotides within the ORF.</TD></TR>
<TR><TD>r:</TD> <TD>Length of the Maximum Run of Non-Insertions observed.</TD></TR>
<TR><TD>s:</TD> <TD>Span of nucleotidies for the Maximum Run of Non-Insertions.</TD></TR>
<TR><TD>zbar</TD> <TD>Posterior Probability of Essentiality.</TD></TR>
<TR><TD>sample</TD> <TD>Comma-Separated list of sample values. Only printed when "-v" flag was printed.</TD></TR>
</table>
<BR>
<BR>



<H4><a name="exec">Running Time</a></H4>

The execution of the software on the representative dataset takes several hours on running on a linux server. In general, this will depend on the number of genes in the genome being analyzed, and the number of of samples desired. The number of samples to be taken can be controlled using the flag "-s". Furthermore, thinning the sample through the "-t" flag will also affect running time as this requires more iterations of the algorithm to achieve the desired effective sample size after thinning. Similarly, the burn-in period (set with the "-b" flag) will also increase the number of iterations required.


<center>&copy; 2012 Michael A. DeJesus</center>
</HTML>
